Astro `user` command documentation (#1829)
astronomer/docs: astro/cli/astro-workspace.md

    ]}>
<TabItem value="astro">

Manage Workspaces on Astro.

## Usage

This command includes two subcommands:

- `astro workspace list`: List all Workspaces.
- `astro workspace switch`: Switch between Workspaces.

## Usage

This command has several subcommands.

### astro workspace create

List all Workspaces.

### astro workspace switch

> Switch between Workspaces. The CLI prompts you to select a Workspace in your current Organization.

Reviewer: ```suggestion
Switch between Workspaces. A prompt instructing you to to select a Workspace in your current Organization appears in the Astro CLI.
```

### astro workspace user add

> Add a new user to your current Astronomer Workspace.

Author: ```suggestion
Add an existing user to your current Astronomer Workspace.
```

> Add a new user to your current Astronomer Workspace. The CLI prompts you for the user's email.

Reviewer: ```suggestion
Add a new user to your current Astronomer Workspace. A prompt asking you to provide the user's email appears in the Astro CLI.
```
**Question:** If the prompt is obvious and expected behavior, is it necessary to describe it?

#### Options

> | `<email>` | Specify the email for the user you want to invite or update. Use only with `astro user update` and `astro user invite`. | Any valid email                                                                                                                             |

Author: ```suggestion
| `<email>` | Specify the email for the user you want to invite or update. | Any valid email                                                                                                                             |
```

> | `--role`  | The user's role in the Workspace. Use only with `astro user update` and `astro user invite`.                         | Possible values are either `WORKSPACE_VIEWER`, `WORKSPACE_OPERATOR`, or `WORKSPACE_OWNER`.|

Author: ```suggestion
| `--role`  | The user's role in the Workspace.                          | Possible values are either `WORKSPACE_VIEWER`, `WORKSPACE_OPERATOR`, or `WORKSPACE_OWNER`.|
```

> | `--role`  | The user's role in the Workspace. Use only with `astro user update` and `astro user invite`.                         | Possible values are either `WORKSPACE_VIEWER`, `WORKSPACE_EDITOR`, or `WORKSPACE_OWNER`. Default is `WORKSPACE_VIEWER` |

Reviewer: ```suggestion
| `--role`  | The user's role in the Workspace. Use only with `astro user update` and `astro user invite`.                         | Valid values are `WORKSPACE_VIEWER`, `WORKSPACE_EDITOR`, or `WORKSPACE_OWNER`. Default is `WORKSPACE_VIEWER` |
```

### astro workspace user update

> Update a user's permissions in your current Astronomer Workspace. The CLI prompts you for the user's email.

Reviewer: ```suggestion
Update a user's permissions in your current Astronomer Workspace.  A prompt asking you to provide the user's email appears in the Astro CLI.
```
**Question**: If the prompt is obvious and expected behavior, is it necessary to describe it?

#### Options

| Option    | Description                                                  | Valid Values                                                                                                                             |
| --------- | ------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------- |

> | `<email>` | Specify the email for the user you want to invite or update. Use only with `astro user update` and `astro user invite`. | Any valid email                                                                                                                             |
> | `--role`  | The user's role in the Workspace. Use only with `astro user update` and `astro user invite`.                         | Valid values are `WORKSPACE_VIEWER`, `WORKSPACE_OPERATOR`, or `WORKSPACE_OWNER`. |

Author: ```suggestion
| `<email>` | Specify the email for the user you want to invite or update.  | Any valid email                                                                                                                             |
| `--role`  | The user's role in the Workspace.                          | Valid values are `WORKSPACE_VIEWER`, `WORKSPACE_OPERATOR`, or `WORKSPACE_OWNER`. |
```

> | `--role`  | The user's role in the Workspace. Use only with `astro user update` and `astro user invite`.                         | Possible values are either `WORKSPACE_VIEWER`, `WORKSPACE_EDITOR`, or `WORKSPACE_ADMIN`. Default is `WORKSPACE_ADMIN` |

Reviewer: ```suggestion
| `--role`  | The user's role in the Workspace. Use only with `astro user update` and `astro user invite`.                         | Valid values are `WORKSPACE_VIEWER`, `WORKSPACE_EDITOR`, or `WORKSPACE_ADMIN`. The default is `WORKSPACE_ADMIN` |
```

> ## Usage
> 
> This command has several subcommands.
> 
> ### astro workspace create
> 
> List all Workspaces.
> 
> ### astro workspace switch
> 
> Switch between Workspaces.
> 
> ### astro workspace user add
> 
> Invite an existing user to your current Astronomer Workspace.
> 
> #### Options
> 
> | `<email>` | The email for the user you want to invite or update. | Any valid email                                                                                                                             |
> | `--role`  | The user's role in the Workspace.                          | Possible values are either `WORKSPACE_VIEWER`, `WORKSPACE_OPERATOR`, or `WORKSPACE_OWNER`.|
> 
> ### astro workspace user update
> 
> Update a user's permissions in your current Astronomer Workspace. The CLI prompts you for the user's email and new role.
> 
> #### Options
> 
> | Option    | Description                                                  | Valid Values                                                                                                                             |
> | --------- | ------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------- |
> | `<email>` | The email for the user you want to invite or update.  | Any valid email                                                                                                                             |
> | `--role`  | The user's role in the Workspace.                          | Valid values are `WORKSPACE_VIEWER`, `WORKSPACE_OPERATOR`, or `WORKSPACE_OWNER`. |
> 
> ### astro workspace user list
> 
> List all users in your current Workspace.
> 
> ### astro workspace user remove
> 
> Remove a user from your current Workspace. 
> 
> #### Options
> 
> | Option    | Description                                                  | Possible Values                                                                                                                             |
> | --------- | ------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------- |
> | `<email>` | The email for the user you want to remove. | Any valid email                                                                                                                             |
> 
> ## Related commands
> 
> - [astro organization](cli/astro-organization.md)

Author: ```suggestion
## Usage

This command has several subcommands.

### `astro workspace create`

List all Workspaces.

### `astro workspace switch`

Switch between Workspaces.

### `astro workspace user add`

Invite an existing user to your current Astronomer Workspace.

#### Options

| `<email>` | The email for the user you want to invite or update. | Any valid email                                                                                                                             |
| `--role`  | The user's role in the Workspace.                          | Possible values are either `WORKSPACE_VIEWER`, `WORKSPACE_OPERATOR`, or `WORKSPACE_OWNER`.|

### `astro workspace user update`

Update a user's permissions in your current Astronomer Workspace. The CLI prompts you for the user's email and new role.

#### Options

| Option    | Description                                                  | Valid Values                                                                                                                             |
| --------- | ------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------- |
| `<email>` | The email for the user you want to invite or update.  | Any valid email                                                                                                                             |
| `--role`  | The user's role in the Workspace.                          | Valid values are `WORKSPACE_VIEWER`, `WORKSPACE_OPERATOR`, or `WORKSPACE_OWNER`. |

### `astro workspace user list`

List all users in your current Workspace.

### `astro workspace user remove`

Remove a user from your current Workspace. 

#### Options

| Option    | Description                                                  | Possible Values                                                                                                                             |
| --------- | ------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------- |
| `<email>` | The email for the user you want to remove. | Any valid email                                                                                                                             |

## Related commands

- [astro organization](cli/astro-organization.md)
```

> | `<email>` | Specify the email for the user you want to invite or update. Use only with `astro user update` and `astro user invite`. | Any valid email                                                                                                                             |

Author: ```suggestion
| `<email>` | The email for the user you want to remove. | Any valid email                                                                                                                             |
```

> - [astro user](cli/astro-user.md)

Author: ```suggestion
- [astro organization](cli/astro-organization.md)
```
